### DISCUSSION TITLE
SSL error for query_staking_summary

### DISCUSSION THREAD

**Discussion by lars20070** [fetchai/cosmpy#397] [Q&A]
I try to retrieve the staked amount of a specific Axelar address with `query_staking_summary` [as discussed in the docu](https://fetch.ai/docs/guides/fetch-network/cosmpy/staking#stake-queries). All gRPC endpoints listed in the [chain registry](https://github.com/cosmos/chain-registry/blob/master/axelar/chain.json#L173) fail with the same SSL error below.
```
Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
```
Back a few months, I was able to connect to at least one of the endpoints. Now all of them fail as above.

Similar problems have been reported [here](https://github.com/grpc/grpc/issues/32076). But I am unsure how further to debug.

**Comment by lars20070** [comment]
Should `cosmpy` update `grpcio` [from 1.59.3](https://github.com/fetchai/cosmpy/blob/main/poetry.lock#L768) to [1.65.0](https://github.com/grpc/grpc/releases/tag/v1.65.0)?

What does `WRONG_VERSION_NUMBER` refer to? Is it a clash between TLS 1.2 and TLS 1.3?

**Comment by kshipra-fetch** [comment]
Hey @lars20070 

Thank you for your query. @pbukva can you please take a look at this one?

**Comment by lars20070** [comment]
@pbukva [Here](https://github.com/fetchai/cosmpy/issues/398#issuecomment-2212435614) the corresponding issue with a minimal example.
